### PR TITLE
Refactor the API/server interface slightly

### PR DESCRIPTION
**Pull request by JulianKniephoff**
See commit messages for more details.

### CODE REVIEW

File: backend/server/src/http/mod.rs [hunk 1]
```diff
@@ -33,10 +34,15 @@ type Request<T = Body> = hyper::Request<T>;
...
     api_root: api::RootNode,
-    api_context: api::Context,
 ) -> Result<()> {
     let assets = Assets::init(&config.assets).await.context("failed to initialize assets")?;
-    let ctx = Arc::new(Context::new(api_root, api_context, assets));
+
+    let db = db::create_pool(&config.db).await
+        .context("failed to create database connection pool (database not running?)")?;
+    db::migrate(&mut *db.get().await?).await
+        .context("failed to check/run DB migrations")?;
```
Comment (LukasKalbertodt) [inline backend/server/src/http/mod.rs:43]: I'm not very thrilled about having this in `http::serve`. Seems wrong to me. I would keep them in `main.rs` and just pass the db pool to `serve`.

File: backend/server/src/http/mod.rs [hunk 2]
```diff
@@ -187,7 +193,11 @@ async fn handle(req: Request<Body>, ctx: Arc<Context>) -> Result<Response> {
 
         // The actual GraphQL API.
         (&Method::GET, "/graphql") | (&Method::POST, "/graphql") => {
-            juniper_hyper::graphql(ctx.api_root.clone(), ctx.api_context.clone(), req).await?
+            juniper_hyper::graphql(
+                ctx.api_root.clone(),
+                Arc::new(api::Context::new(ctx.db.get().await?).await?),
+                req
+            ).await?
```
Comment (LukasKalbertodt) [inline backend/server/src/http/mod.rs:200]: Trailing comma after `req`. Or pull out the second argument as `let db_conn = ...;` and write the `graphql` call in one line.

**Review by LukasKalbertodt — CHANGES_REQUESTED** [review]
I guess it makes sense to use a single DB connection for one API request. It's a shame I don't have benchmarks in place to test the impact of this change. But it should be fine. A few inline comments.

File: backend/server/src/http/mod.rs (hunk 2 above)
Comment (JulianKniephoff) [reply backend/server/src/http/mod.rs]: Moved it back. :)

File: backend/server/src/http/mod.rs (hunk 1 above)
Comment (JulianKniephoff) [reply backend/server/src/http/mod.rs]: Moved it back. :)

**Comment by LukasKalbertodt** [comment]
I guess this PR is superseded by #236 and can be closed, right?

**Comment by JulianKniephoff** [comment]
This is probably rendered mostly obsolete by #236.

**Pull request closed without merging**